src/ir_def/rename.ml

```diff
@@ -62,10 +62,10 @@ and exp' rho = function
   | DeclareE (i, t, e)  -> let i',rho' = id_bind rho i in
                            DeclareE (i', t, exp rho' e)
   | DefineE (i, m, e)   -> DefineE (id rho i, m, exp rho e)
-  | FuncE (x, s, c, tp, p, ts, e) ->
+  | FuncE (x, s, c, tp, p, ts, closure, e) ->
```

Comment:
I closure records the set of free variables, don't those need to be renamed consistently too? But if they are, doesn't that break the names of stable variables?

Reply:
Yes, consistent renaming would be needed, maybe by indirection or a symbol table in longer run. When renaming closure set would be consistently updated too.